### DISCUSSION TITLE
application  registration/ Service Principal

### DISCUSSION THREAD

**Discussion by pattisanta** [microsoftgraph/msgraph-bicep-types#189] [General]
Hi All,
I able to create a service principal successfully with bicep. what I am looking to do., is to add owner directly on the application  registration/ Service Principal  via bicep for some reason I can't find  in bicep  I have attached a screenshot for visual on what it looks like in entra. 
<img width="345" alt="test" src="https://github.com/user-attachments/assets/a90fa7b9-0e13-413a-a2e2-95a24d5b495e">

Thanks,
Patti

**Comment by dkershaw10** [comment]
Hi @pattisanta. Unfortunately, the owners property is not supported right now on the applications or service principals types.  It is something in our backlog, but unsure when we'll get to this.  It's tracked in #134 although I may create a new item for this, as that issue tracks multiple problems, including what you are asking for.

Can you provide the scenario for why you need the owners property on these types please?

**Reply by pattisanta** [reply]
@dkershaw10  Thanks for replying. We add users the owner role to our applications/service principals  because, we a have  owners get notified when their secret or cert is  near expiry or expired.  

Patti

**Reply by pattisanta** [reply]
@dkershaw10  I've seen updates that  we can now add owners to the  application registration/ Service Principal. I don't any new property in the extension microsoftGraphV1_0 . Can you tell me what I should be looking for?

Patti

**Reply by dkershaw10** [reply]
@pattisanta The updates you may have been seeing are all about **automatically** adding the creating security principal as an owner (user if there is a sign-in user OR service principal if this is a zero-touch deployment with no signed-in user).  We had a bug in that logic, which I believe is now fixed.

This is different from explicitly specifying the owners in the resource itself, which is what you are after.  Unfortunately this capability is still in our backlog.  The only workaround here is to use a deployment script (which is definitely less than ideal).

Created an explicit enhancement request for this feature ask (to separate it from previous issues): #194
